walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -22,6 +22,19 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    int nameBegin = 0;
+    int nameEnd = name.length() - 1;
+    while (nameBegin < nameEnd && name.charAt(nameBegin) == ' ') {
+        ++nameBegin;
+    }
+    while (nameBegin < nameEnd && name.charAt(nameEnd) == ' ') {
+        --nameEnd;
+    }
+
+    if (nameBegin >= nameEnd) {
+        return "Hello!";
+    } else {
+        return "Hello " + name.substring(nameBegin, nameEnd + 1);
+    }
```

Comment:
Could you try to use `java.lang.String.trim()` method instead of customizing the string trimming logic? 

https://www.tutorialspoint.com/java/lang/string_trim.htm